mavlink/modules/flight_controller.py

```diff
@@ -95,3 +95,21 @@ def get_home_location(self, timeout: float) -> "tuple[bool, drone_odometry.Drone
             return False, None
 
         return True, location
+
+    def is_drone_destination_final_waypoint(self) -> "tuple[bool, bool | None]":
+        """
+        Returns whether the drones destination is the final waypoint in the mission.
+        """
+        waypoint_count = self.drone.commands.count
+        current_waypoint = self.drone.commands.next
+
+        if waypoint_count < 0 or current_waypoint < 0:
+            return False, None
+
+        if waypoint_count == 0:
+            return True, False
+        else:
+            if current_waypoint == waypoint_count:
+                return True, True
+            else:
+                return True, False
```

Comment:
Unnecessarily complex. Replace with:
```py
return True, (current_waypoint == waypoint_count)
```
Parantheses optional, but I prefer because it makes it easier to read.